docs/theories/sets-and-relations.rst

```diff
@@ -0,0 +1,201 @@
+Theory Reference: Sets and Relations
+====================================
+
+Finite Sets
+-----------
+
+cvc5 supports the theory of finite sets.
+The simplest way to get a sense of the syntax is to look at an example:
+
+.. api-examples::
+    ../../examples/api/cpp/sets.cpp
+    ../../examples/api/python/sets.py
+    ../../examples/api/smtlib/sets.smt2
```

Comment:
Do we want to see SMT-LIB by default here?

Reply:
I used the same order as for the others, and I'd rather be consistent over all examples in terms of order. I don't see a reason why we should prefer smt2 here.